File: Harjoitus_1/exercise1_4.py
def minuutitTunneiksi(minuutit):
    tunnit = minuutit // 60
    minuutit = minuutit % 60
    return (tunnit, minuutit)

def main():
    try:
        minuutit = int(input("Anna minuuttien määrä: "))
    except ValueError:
        print("Virheellinen syöte. Anna kokonaisluku.")
        return
    if minuutit < 0:
        print("Minuuttien määrä ei voi olla negatiivinen.")
        return
    tunnit, minuutit = minuutitTunneiksi(minuutit)
    if tunnit == 0:
        print(f"{minuutit} minuuttia")
    elif minuutit == 0:
        print(f"{tunnit} tuntia")
    else:
        print(f"{tunnit} tuntia ja {minuutit} minuuttia")

if __name__ == "__main__":
    main()

File: cpp/Harjoitus_1/exercise1_4.cpp
#include <iostream>
#include <limits>
#include <iomanip>
#include <utility>

int min;
int hr;

int main() {
    std::cout << "Anna minuutit:\n> ";
    while (!(std::cin >> min)){
        std::cout << "Väärä syöte. Anna kokonaisluku:\n> ";
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(),'\n');
    }
    if (min <= 0) {
        std::cout << "Minuutit eivät voi olla yhtä kuin tai vähemmän kuin 0\n";
        min = 0; 
        main();
    }
    hr = min / 60;
    min = min % 60;
    std::cout << "Tunnit: " << hr << "\nMinuutit: " << min << '\n';
}




